clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -4012,7 +4012,13 @@ LangAS CodeGenModule::getStringLiteralAddressSpace() const {
     //   const char *getLiteral() n{
     //     return "AB";
     //   }
-    return LangAS::opencl_private;
+    // because there is a addressspacecast to generic address space in IR,
+    // but adressspacecast from constant to generic forbitten because of
+    // constant address space is not part of generic address space.
+    // The private adress space doesn't suit here because a IR is translated
+    // in SPIRV in SYCLIsDevice mode, and all global Value shouldn't
+    // be private in IR for rigth translation to SPIRV.
```

Comment:
```suggestion
    // Use global address space to avoid illegal casts from constant to generic. 
    // Private address space is not used here because in SPIR-V global values
    // cannot have private address space.
```